monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,481 @@
+            flight_points_with_altitude = []
+            x, y = buffered_path.exterior.coords.xy
+
+            for coord in range(0, len(x)):
+                cur_coord = coord
+                next_coord = coord + 1 
+                next_coord = 0 if next_coord == len(x) else next_coord
+                adjacent_points = [Point(x[cur_coord], y[cur_coord]), Point(x[next_coord], y[next_coord])]
+                flight_speed, bearing  = self.generate_flight_speed_bearing(adjacent_points=adjacent_points, delta_time_secs= 1)
+                
+                flight_points_with_altitude.append(FlightPoint(lat = y[coord], lng = x[coord], alt = altitude, speed = flight_speed, bearing = bearing))
+
+            all_grid_cell_tracks.append(GridCellFlight(bounds = grid_cell, track = flight_points_with_altitude))
+
+        self.grid_cells_flight_tracks = all_grid_cell_tracks
+
+    def make_json_compatible(self, struct: Any) -> Any:
+        if isinstance(struct, tuple) and hasattr(struct, '_asdict'):
+            return {k: self.make_json_compatible(v) for k, v in struct._asdict().items()}
+        elif isinstance(struct, dict):
+            return {k: self.make_json_compatible(v) for k, v in struct.items()}
+        elif isinstance(struct, str):
+            return struct
+        try:
+            return [self.make_json_compatible(v) for v in struct]
+        except TypeError:
+            return struct
+
+    def generate_rid_state(self, duration=180):
+        '''
+
+        This method generates rid_state objects that can be submitted as flight telemetry
+
+
+        '''
+        all_flight_telemetry = {}
+        flight_track_details = {}  # Develop a index of flight length and their index
+        # Store where on the track the current index is, since the tracks are circular, once the end of the track is reached, the index is reset to 0 to indicate beginning of the track again.
+        flight_current_index = {}
+        # Get the number of flights
+        num_flights = len(self.grid_cells_flight_tracks)
+        time_increment_seconds = 1 # the number of seconds it takes to go from one point to next on the track
+        now = arrow.now()
+        now_isoformat = now.isoformat()
+        for i in range(num_flights):
+            flight_positions_len = len(self.grid_cells_flight_tracks[i].track)
+            
+            # in a circular flight pattern increment direction
+            angle_increment = (360 / flight_positions_len)
+            
+            # the resolution of track is 1 degree minimum
+            angle_increment = 1.0 if angle_increment == 0.0 else angle_increment
+
+            if i not in flight_track_details:
+                flight_track_details[i] = {}
+            flight_track_details[i]['track_length'] = flight_positions_len
+            
+            flight_current_index[i] = 0
+            all_flight_telemetry[i]= {}
+            all_flight_telemetry[i]['states'] = []
+
+        for j in range(duration):
```

Comment:
nit: Could be shortened to remove the conditional check with

```python
timestamp = now
for j in range(duration):
    timestamp = timestamp.shift(seconds=time_increment_seconds)
[...]
```

Reply:
OK thanks, yes I have now fixed it per your recommendation.